### PR TITLE
Pre-commit hook instructions for windows using hardlinks

### PR DESCRIPTION
**Pull request by whatnick**
## Description
Set-up pre-commit hook hardlink on windows.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -15,6 +15,14 @@ This can be automated by setting up the pre-commit hook properly.
 
     ln -s ../../scripts/prepare-commit.sh .git/hooks/pre-commit
 
+Windows equivalent for a symlink using PowerShell in elevated priviledges mode from the .git\hooks folder. This will create hardlink which can be used from git-bash shell.
```
Comment (m-kuhn) [inline .github/CONTRIBUTING.md]: I'm by no means a Windows expert, so sorry if the question is trivial.
Why is there a requirement for elevated privileges to do something in a user folder?

(Sidenote, there is a typo in privileges)

Comment (jef-n) [reply .github/CONTRIBUTING.md]: @whatnick Why the powershell reference?  I'd use cmd too (if cygwin is unavailable).
@m-kuhn on windows links are privileged business.

**Comment by whatnick** [comment]
Just a privileged shell will do. Powershell has additional niceness.

**Comment by jef-n** [comment]
@whatnick,

On Tue, 14. Nov 2017 at 12:00:50 +0000, Tisham Dhar wrote:
> Just a privileged shell will do. Powershell has additional niceness.

I just meant the inconsistency: you mention powershell, but use cmd.
BTW apparently powershell doesn't need cmd and can do it with New-Item.

Looks like I don't have to resort to using mklink from cmd in cygwin either :
https://stackoverflow.com/questions/18654162/enable-native-ntfs-symbolic-links-for-cygwin/18659632#18659632

(I usually let links in qgis' user plugin dir point at my plugins' source)

**Comment by whatnick** [comment]
I cannot find a reasonable next step for this where I can reuse gitperl with Yaml::Tiny. Suggestions appreciated.

**Comment by nyalldawson** [comment]
I'd suggest updating the instructions to make use of bash for windows... that's the only way I've found to get this working reliably under windows.

**Comment by jef-n** [comment]
> I'd suggest updating the instructions to make use of bash for windows... that's the only way I've found to get this working reliably under windows.

You mean cygwin's bash?

**Comment by whatnick** [comment]
I believe Nyall means Windows Linux Subsystem i.e. Canonical + Microsoft
bash.

**Comment by m-kuhn** [comment]
What's the status of this?
Something to improve and merge?

**Comment by nyalldawson** [comment]
@m-kuhn let's close? I think given there's debate about this approach and lack of consensus on the correct approach we should close and wait for a refined pr.

**Pull request closed without merging**